pkg/disttask/framework/taskexecutor/state_transform.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package taskexecutor
+
+import (
+	"github.com/pingcap/tidb/pkg/disttask/framework/proto"
+)
+
+// VerifySubtaskStateTransform verifies whether the subtask state transform is valid.
+func VerifySubtaskStateTransform(from, to proto.SubtaskState) bool {
```

Comment:
this check won't work when subtask state change in between

Reply:
Yes, I will add it. This function only used during development